### DISCUSSION TITLE
Recent Updates!

### DISCUSSION THREAD

**Discussion by djyt** [djyt/cannonball#73] [General]
Hello, 

Firstly, an apology for ignoring issues on here for the past 6 years or so. I haven't been actively working on CannonBall in recent years and for a long time didn't even have the code compiling on my PC. Due to starting a family and life, I found my hobby time decreased substantially. Plus I tend to rotate through a wide range of hobbies and projects, and randomly work on what I'm interested in. If I'm not having fun, I move on. This spans music production, coding and repairing arcade hardware. 

In the meantime, I worked on [OutRun Enhanced v2](http://reassembler.blogspot.com/2020/04/outrun-enhanced-edition-201.html) and [AfterBurner Enhanced](http://reassembler.blogspot.com/2017/04/afterburner-2-enhanced-edition.html) for arcade hardware which are both awesome. Try them out if you are lucky enough to have an original cab. 

There will soon be a (small) update to support the SmartyPi arcade hardware interface which is really exciting. [More details here.](http://www.ukvac.com/forum/smartypi-emulation-pcb-replacement-sega-edition_topic381677.html)

**In recent months I've completed some general housekeeping to the codebase:**

- Greatly simplified the cmake build system. It had become too convoluted and even I didn't understand it after 6 years. Right now, I've only checked in cmake files for Windows and the Pi 4. To prove the portability of the system it literally took me 30 mins to get the build running on the Pi.

- Updated the codebase to work with Visual Studio 2019 and a more recent Boost library. 

- Removed CannonBoard support and replaced with SmartyPi support

- Removed legacy SDL 1 support to clean up the code

- Fixed bugs (audio speed issue and the time trial crash)

- I've also setup notifications so I can see when comments and issues are added.

I will aim to make further improvements to CannonBall beyond these in the coming months. There is plenty of work still to be done, which won't result in anything exciting from a feature-set perspective. There is also an unending array of features I can dream of. 

Best Wishes,
Chris.

**Comment by baritonomarchetto** [comment]
I am so happy one of my favourite projects is still ongoing!
Thanks Chris!
